website/content/en/blog/log-namespacing.md

```diff
@@ -0,0 +1,155 @@
+
+Every source also has a `log_namespace` config option. This will override the global setting,
+so you can try out Log Namespacing on individual sources.
+
+The following example enables the `log_namespace` feature globally, then disables it for a single
+source.
+
+```toml
+schema.log_namespace = true
+
+[sources.input_with_log_namespace]
+type = "demo_logs"
+format = "shuffle"
+lines = ["input_with_log_namespace"]
+interval = 1
+
+[sources.input_without_log_namespace]
+type = "demo_logs"
+format = "shuffle"
+lines = ["input_without_log_namespace"]
+interval = 1
+log_namespace = false
+
+[sinks.console]
+type = "console"
+inputs = ["input_with_log_namespace", "input_without_log_namespace"]
+encoding.codec = "json"
+
+```
+
+## How It Works
+
+
+### Data Layout
+
+There are three distinct types of data that Vector handles.
+
+(Examples are from the `http_client` source)
+- Event Data: The decoded event data. (eg: the decoded HTTP body)
+- Source Metadata: Metadata provided by the source of the event. (eg: the HTTP headers)
+- Vector Metadata: Metadata provided by Vector. (eg: the time when Vector received the event)
+
+#### Without Log Namespacing
+All three of these are placed at the root of the event. The exact layout depends on the source,
+some fields are configurable, and the [global log schema] can change the name / location of some
+fields.
+
+Example log event from the `datadog_agent` source (with the JSON decoder)
+
+```json
+{
+  "ddsource": "vector",
+  "ddtags": "env:prod",
+  "hostname": "alpha",
+  "foo": "foo field",
+  "service": "cernan",
+  "source_type": "datadog_agent",
+  "bar": "bar field",
+  "status": "warning",
+  "timestamp": "1970-02-14T20:44:57.570Z"
+}
+```
+
+#### With Log Namespacing
+When enabled, the layout of this data is well-defined and consistent.
+
+Event Data (and _only_ Event Data) is placed at the root of the event (eg: `.`).
+Source metadata is placed in event metadata, prefixed by the source name. (eg: `%datadog_agent`)
+Vector metadata is placed in event metadata, prefixed by `vector`. (eg: `%vector`)
+
+Generally sinks will only send the event data. If you want to include any metadata fields,
+it's recommended to use a [remap] transform to add data to the event as needed.
+
+It's important to note that previously the type of an event (`.`) would always be an object
+with fields. Now it is possible for event to be any type, such as a string.
+
+Example event from the `datadog agent logs` source. (same data as the example above)
+
+Event root (`.`)
+```json
+{
+  "foo": "foo field",
+  "bar": "bar field"
+}
+```
+
+Source metadata fields (`%datadog_agent`)
+
+```json
+{
+  "ddsource": "vector",
+  "ddtags": "env:prod",
+  "hostname": "alpha",
+  "service": "cernan",
+  "status": "warning",
+  "timestamp": "1970-02-14T20:44:57.570Z"
+}
+```
+
+Source vector fields (`%vector`)
+```json
+{
+  "source_type": "datadog_agent",
+  "ingest_timestamp": "1970-02-14T20:44:58.236Z"
+}
+```
```

Comment:
To really drive home the UX here, maybe add a small VRL script accessing/moving fields from the different namespaces.